astronomy/orbit_analysis_test.cpp

```diff
@@ -0,0 +1,509 @@
+    return files;
+  }
+};
+
+struct SP3Orbit {
+  StandardProduct3::SatelliteIdentifier satellite;
+  SP3Files files;
+};
+
+}  // namespace
+
+class OrbitAnalysisTest : public ::testing::Test {
+ protected:
+  OrbitAnalysisTest()
+      : earth_1957_(RemoveAllButEarth(SolarSystem<ICRS>(
+            SOLUTION_DIR / "astronomy" / "sol_gravity_model.proto.txt",
+            SOLUTION_DIR / "astronomy" /
+                "sol_initial_state_jd_2436116_311504629.proto.txt"))),
+        ephemeris_(earth_1957_.MakeEphemeris(
+            /*accuracy_parameters=*/{/*fitting_tolerance=*/1 * Milli(Metre),
+                                     /*geopotential_tolerance=*/0x1p-24},
+            Ephemeris<ICRS>::FixedStepParameters(
+                SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                                   Position<ICRS>>(),
+                /*step=*/1 * JulianYear))),
+        earth_(*earth_1957_.rotating_body(*ephemeris_, "Earth")) {}
+
+  // Returns a GCRS trajectory obtained by stitching together the trajectories
+  // of |sp3_orbit.satellites| in |sp3_orbit.files|.
+  not_null<std::unique_ptr<DiscreteTrajectory<GCRS>>> EarthCentredTrajectory(
+      SP3Orbit const& sp3_orbit) {
+    BodyCentredNonRotatingDynamicFrame<ICRS, GCRS> gcrs{ephemeris_.get(),
+                                                        &earth_};
+    BodySurfaceDynamicFrame<ICRS, ITRS> itrs{ephemeris_.get(), &earth_};
+
+    auto result = make_not_null_unique<DiscreteTrajectory<GCRS>>();
+    for (auto const& file : sp3_orbit.files.names) {
+      StandardProduct3 sp3(
+          SOLUTION_DIR / "astronomy" / "standard_product_3" / file,
+          sp3_orbit.files.dialect);
+      auto const& orbit = sp3.orbit(sp3_orbit.satellite);
+      CHECK_EQ(orbit.size(), 1);
+      auto const& arc = *orbit.front();
+      for (auto it = arc.Begin(); it != arc.End(); ++it) {
+        ephemeris_->Prolong(it.time());
+        result->Append(
+            it.time(),
+            gcrs.ToThisFrameAtTime(it.time())(
+                itrs.FromThisFrameAtTime(it.time())(it.degrees_of_freedom())));
+      }
+    }
+    return result;
+  }
+
+  std::pair<OrbitalElements, OrbitRecurrence> ElementsAndRecurrence(
+      SP3Orbit const& orbit) {
+    auto const elements =
+        OrbitalElements::ForTrajectory(*EarthCentredTrajectory(orbit),
+                                       earth_,
+                                       MasslessBody{}).ValueOrDie();
+    auto const recurrence =
+        OrbitRecurrence::ClosestRecurrence(elements.nodal_period(),
+                                           elements.nodal_precession(),
+                                           earth_,
+                                           /*max_abs_Cᴛₒ=*/100);
+    return {elements, recurrence};
+  }
+
+  SolarSystem<ICRS> earth_1957_;
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> ephemeris_;
+  RotatingBody<ICRS> const& earth_;
```

Comment:
No oblateness?

Reply:
We have no need to know about the properties of the oblateness here.